### PR TITLE
fix: add gradient gathering after input split in Context Parallel

### PR DESCRIPTION
**Pull request by baoleai**

### CODE REVIEW

File: torchacc/ops/context_parallel/utils.py [hunk 1]
```diff
@@ -155,28 +155,54 @@ def flash_attention(
...
+
+    @staticmethod
+    def forward(ctx, tensor, seq_dim, process_group):
+        ctx.process_group = process_group
+        ctx.seq_dim = seq_dim
+        return _split(tensor, seq_dim, process_group)
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        return _gather(grad_output, ctx.seq_dim, ctx.process_group), None, None
```
Comment (yitongh) [inline torchacc/ops/context_parallel/utils.py:212]: Do we need to divide by world_size here?

Comment (baoleai) [reply torchacc/ops/context_parallel/utils.py:212]: added the argument 'grad_scale' to enable scale by world_size

File: torchacc/ops/context_parallel/utils.py [hunk 2]
```diff
@@ -155,68 +155,110 @@ def flash_attention(
...
+                                  grad_scale=None):
+    """ Split the input tensor along sequence dimension during forward
+        and gather the input tensor during backward, which are parallelized
+        across GPUs in a context parallel group.
+    Args:
+        tensor (torch.Tensor): [batch_size, seqlen * cp_size, nheads, headdim].
+        seq_dim (int): The dimension for split and gather.
+        process_group (torch.distributed.ProcessGroup, optional): The context parallel group.
+        grad_scale (str, optional): The gradient scale. 'up' or 'down'. 'up' means the
+        gradient will be multiplied by the size of process group, and 'down' means the
```
Comment (yitongh) [inline torchacc/ops/context_parallel/utils.py]: indent？

Comment (baoleai) [reply torchacc/ops/context_parallel/utils.py]: done

**Review by yitongh — APPROVED** [review]

**Pull request merged**